src/DelegateRegistry.sol

```diff
@@ -104,199 +100,375 @@ contract DelegateRegistry is IDelegateRegistry {
-        bool value,
-        IDelegateRegistry.DelegationType type_,
-        address vault,
-        address contract_,
-        uint256 tokenId,
-        uint256 balance,
-        bytes32 data
-    ) internal {
-        if (value) {
-            vaultDelegationHashes[vault].add(delegationHash);
-            delegateDelegationHashes[delegate].add(delegationHash);
-            delegationInfo[delegationHash] =
-                DelegationInfo({vault: vault, delegate: delegate, type_: type_, contract_: contract_, tokenId: tokenId, balance: balance, data: data});
-        } else {
-            vaultDelegationHashes[vault].remove(delegationHash);
-            delegateDelegationHashes[delegate].remove(delegationHash);
-            delete delegationInfo[delegationHash];
+    /**
+     * ----------- Consumable -----------
+     */
+
+    /// @inheritdoc IDelegateRegistry
+    function checkDelegateForAll(address delegate, address vault, bytes32 rights) public view override returns (bool valid) {
+        bytes32 location = _computeDelegationLocation(_computeDelegationHashForAll(delegate, "", vault));
+        valid = _loadDelegationAddress(location, StoragePositions.vault) == vault;
+        if (rights != "" && !valid) {
+            location = _computeDelegationLocation(_computeDelegationHashForAll(delegate, rights, vault));
+            valid = _loadDelegationAddress(location, StoragePositions.vault) == vault;
         }
     }
 
-    /// @dev Helper function to compute delegation hash for wallet delegation
-    function _computeDelegationHashForAll(address vault, address delegate, bytes32 data) internal pure returns (bytes32) {
-        return keccak256(abi.encode(delegate, vault, data));
+    /// @inheritdoc IDelegateRegistry
+    function checkDelegateForContract(address delegate, address vault, address contract_, bytes32 rights) public view override returns (bool valid) {
+        bytes32 location = _computeDelegationLocation(_computeDelegationHashForContract(contract_, delegate, "", vault));
+        valid = checkDelegateForAll(delegate, vault, "") || _loadDelegationAddress(location, StoragePositions.vault) == vault;
+        if (rights != "" && !valid) {
+            location = _computeDelegationLocation(_computeDelegationHashForContract(contract_, delegate, rights, vault));
+            valid = checkDelegateForAll(delegate, vault, rights) || _loadDelegationAddress(location, StoragePositions.vault) == vault;
+        }
     }
 
-    /// @dev Helper function to compute delegation hash for contract delegation
-    function _computeDelegationHashForContract(address vault, address delegate, address contract_, bytes32 data) internal pure returns (bytes32) {
-        return keccak256(abi.encode(delegate, vault, contract_, data));
+    /// @inheritdoc IDelegateRegistry
+    function checkDelegateForERC20(address delegate, address vault, address contract_, bytes32 rights) external view override returns (uint256 balance) {
```

Comment:
Still thinking about whether this (& check1155) should diverge from the others and return a uint256, or expect the `amount` as a param and simply return a `bool` validity result